core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -277,8 +277,8 @@ private[spark] class RestSubmissionClient(master: String) extends Logging {
   private def getBaseUrl(master: String): String = {
     var masterUrl = master
     supportedMasterPrefixes.foreach { prefix =>
-      if (master.startsWith(prefix)) {
-        masterUrl = master.stripPrefix(prefix)
+      if (masterUrl.startsWith(prefix)) {
+        masterUrl = masterUrl.stripPrefix(prefix)
```

Comment:
Actually, why is this an issue? If we assume that the master URL matches only one of the `supportedMasterPrefixes`, then this change doesn't actually do anything, since we set `masterUrl = master` up there.